P8/DataConverters/ShapeletHistogramConverter/FeatureExtraction/src/core/WindowGeneration.h

```diff
@@ -5,52 +5,81 @@
 #include "IO/Logger.h"
 
 namespace WindowGeneration {
-    [[nodiscard]] static std::vector<Series> GenerateWindows(const Series &series, uint length) {
+    /// <summary>
+    /// Generates a vector of sub-series out of a series, based on a specific window size.
+    /// </summary>
+    /// <param name="series">A vector of doubles</param>
+    /// <param name="length">An integer, representing the length of each window to be made</param>
+    /// <returns>A vector of sub-series</returns>
+    [[nodiscard]] static std::vector<Series> GenerateWindowsOfLength(const Series &series, uint length) {
         std::vector<Series> windows;
 
         if (series.empty() || length == 0)
             return windows;
 
-        for (uint i = 0; i + length <= series.size(); ++i) {
-            Series window;
-            const double yOffset = series[i]; // Always start a window at 0
-            for (uint offset = 0; offset < length; ++offset)
-                window.push_back(series[i + offset] - yOffset);
-            windows.push_back(window);
+        for (uint i = 0; i < series.size(); i++) {
+            if (i + length <= series.size()) {
+                Series window;
+                for (uint j = 0; j < length; j++)
+                    window.push_back(series[i + j]);
```

Comment:
I'm not sure it is a good idea to remove window y-axis normalization.

Reply:
Because it didn't work as intended before
Idk if it's only with window sized of 1,but I noticed all the windows just became the first window

Reply:
It is the intended behavior, as otherwise the position the shapelet was found in matters.
e.g. [0,1] is different than [1,2] although the visually it's the same
Whereas with prior behavior it would be [0,1] and [1,2] -> [0,1]

A side effect of this is that the minimum window size is 2, where a window size of 1 would result in all 0's.

Also, as a sidenote try running experiments with these changes. I saw a decrease in accuracy of 70% to 50%

Reply:
I mean, i can add it back again
But i see that this can really be an issue with time series that have the same "curves" between classes, but at different y axis values.
I suspect it might also be why we get quite bad results in the arrowhead dataset? Just a guess, but the data in that all pretty much have the same shapes